### PR TITLE
Add handler delegates

### PR DESCRIPTION
**Pull request by FreePhoenix888**
- Remove unused code and update XML Documentation
- Repair XML Documentation
- Add handler delegates
- Use WriteHandler
- Use ReadHandler

### CODE REVIEW

**Review by Konard — CHANGES_REQUESTED** [review]

File: csharp/Platform.Data/ReadHandler.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+using System.Collections.Generic;
+
+namespace Platform.Data;
+
+public delegate TLink ReadHandler<TLink>(IList<TLink> restriction);
```
[the remark this answers is not in the document]
Comment (FreePhoenix888) [reply csharp/Platform.Data/ReadHandler.cs:5]: We'll deviate from style this way

[the remark this answers is not in the document]
Comment (uselessgoddess) [reply csharp/Platform.Data/ReadHandler.cs:5]:
```suggestion
public delegate TLink ReadHandler<TLink>(IList<TLink>);
```

File: csharp/Platform.Data/WriteHandler.cs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+using System.Collections.Generic;
+
+namespace Platform.Data;
+
+public delegate TLink WriteHandler<TLink>(IList<TLink> restriction, IList<TLink> substitution);
```
[the remark this answers is not in the document]
Comment (FreePhoenix888) [reply csharp/Platform.Data/WriteHandler.cs:5]: We'll deviate from our style this way, will not we?

File: csharp/Platform.Data/ReadHandler.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (FreePhoenix888) [reply csharp/Platform.Data/ReadHandler.cs:5]: @uselessgoddess amazing

**Review by Konard — APPROVED** [review]

**Pull request merged**